src/components/UserProfile/UserSettings/UserNotificationSettings/index.tsx

```diff
@@ -32,6 +36,13 @@ const UserNotificationSettings: React.FC = () => {
     user ? `/api/v1/user/${user?.id}/settings/notifications` : null
   );
 
+  const UserNotificationSettingsSchema = Yup.object().shape({
+    discordId: Yup.string().matches(
+      /^\d{18}$/,
+      intl.formatMessage(messages.validationDiscordId)
+    ),
+  });
```

Comment:
I think this will block saving for people who never fill it in. The default value is null for `discordId` which I believe will fail the string check here.

Reply:
We had a similar issue where people couldn't edit users without setting a displayname.